src/web100srv.c

```diff
@@ -1642,12 +1646,16 @@ ndtchild *spawn_new_child(int listenfd, SSL_CTX *ssl_context) {
         accept_errno = errno;
         log_println(1, "accept() on ctlsockfd failed");
         log_println(1, "Error was: %s (%d)", strerror(accept_errno), accept_errno);
-        if (accept_errno != EINTR) return NULL;
+        if (accept_errno != EINTR) {
+          log_println(1, "accept() failed unrecoverably. Child terminating.");
+          exit(-1);
+        }
       }
     } while (accept_errno == EINTR);
     if (cli_addr_len > sizeof(meta.c_addr)) {
```

Comment:
Optional: this check and block should not be a runtime check. The types are declared to be the same, if they were different sizes then this would be a compile-time error.